tasks/automatedClaim.ts

```diff
@@ -16,14 +29,49 @@ task('automatedClaim', 'Claim all rewards for a season')
     const ClaimFacet = await ethers.getContractAt('ClaimFacet', diamondAddress)
     // Load all Deposit events
     const filter = DepositFacet.filters.Deposit(seasonId, null, null)
-    const depositEvents = await DepositFacet.queryFilter(filter)
+    /*
+     * Query events for evey 10,000 blocks since FROM_BLOCK
+     * using a limiter to avoid rate limiting from RPC provider
+     * @dev Ankr has a rate limit of 30 requests per second
+     */
+    const lastBlock = await ethers.provider.getBlockNumber()
+    const result = await limiter.schedule(async () => {
+      let depositEvents: DepositEvent[] = []
+      for (let i = FROM_BLOCK[network.name]; i < lastBlock; i += 10_000) {
+        const toBlock = i + 10_000 > lastBlock ? 'latest' : i + 10_000
+        console.log(`Querying events from block ${i} to ${toBlock}`)
+        const events = await DepositFacet.queryFilter(filter, i, toBlock)
+        console.log(`Found ${events.length} events`)
+        depositEvents = depositEvents.concat(events)
+      }
+      console.log(`Found ${depositEvents.length} total events`)
+      return depositEvents
+    })
+
     // Get all depositors
-    const depositors = depositEvents.map((event) => event.args?.user)
+    if (dryRun === 'true') return // If dry run, return early
+    const depositors = result.map((event) => event.args?.user)
     const uniqueDepositors = [...new Set(depositors)]
-    // Slice the array into chunks of 100
-    const chunkedDepositors = chunk(uniqueDepositors, 100)
-    // Claim for each chunk
-    for (const chunk of chunkedDepositors) {
-      await (await ClaimFacet.automatedClaim(seasonId, chunk)).wait(3)
+
+    try {
+      // Slice the array into chunks of 100
+      const chunkedDepositors = chunk(uniqueDepositors, 100)
+      // Claim for each chunk
+      for (const chunk of chunkedDepositors) {
+        await (await ClaimFacet.automatedClaim(seasonId, chunk)).wait(3)
+      }
+    } catch (e) {
+      console.error(e)
+      throw e
     }
+
+    /*
+     * TODO: implement starting a new season right after claiming
+     * 1. Calculate new season rewards amount (VAPE)
+     * 2. Calculate new season start and end timestamps
+     * 3. Call EmmissionsManager to mint VAPE
+     * 4. Call Diamond to start new season
```

Comment:
hey @parth-15 can you confirm this?

Reply:
Hey @mejiasd3v , this is not correct. The correct order is as follows:
* Calculate new season rewards amount (VAPE)
* Calculate new season start and end timestamps
* Call Diamond to start new season
* Call diamond to mint VAPE through emissions manager(function name is `claimTokensForSeason`)